### PR TITLE
Docs: Add how to get agent debug logs.

### PR DESCRIPTION
**Pull request by jcvenegas**
This patch adds documentation on how to get
agent debug logs.

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by mcastelino** [comment]
LGTM

File: docs/debug-agent.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Get Clear Containers Agent debug logs
+
+When a new pod is created Clear Containers runtime launch a special VM which
+uses a VM-agent. The VM-agent allows to spawn processes on behalf of
+pod/container(s) running inside the virtual machine. 
+
+From Clear Containers 3.0-beta release a new agent is used,
+[Clear Containers Agent](https://github.com/clearcontainers/agent):
+
+The Clear Containers Agent  relies heavily on libcontainer used by runc
```
Comment (devimc) [inline docs/debug-agent.md]: two spaces between Agent  and relies

File: docs/debug-agent.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+(standard to run containers on bare metal).
+
+To provide a debug log of any agent activity on a guest, the Clear Containers
+agent sends logs through a qemu serial console` that are collected by `cc-proxy
```
Comment (devimc) [inline docs/debug-agent.md]: ` that are collected by `  ->  `that are collected by`

File: docs/debug-agent.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+and shown in its logs. By default, this agent logs are not collected by 
+[cc-proxy](https://github.com/clearcontainers/proxy) but can be enabled with a
+simple change to service file.
+
+1- Add `-log debug` option to  `cc-proxy`  
+
+```
+$sudo sed -i \
```
Comment (devimc) [inline docs/debug-agent.md]: you could create a configuration file in /etc/systemd/system/cc-proxy.service.d to override the ExecStart line

File: docs/debug-agent.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+```
+2- Run a container to generate agent debug logs
+
+```
+sudo docker run -ti busybox true
+```
+3- Filter agent debug logs from cc-proxy logs
+
+```
+sudo journalctl -u cc-proxy | grep source=qemu
```
Comment (devimc) [inline docs/debug-agent.md]: Q. does ```source``` alway is ```qemu``` ?

Comment (jcvenegas) [inline docs/debug-agent.md]: @devimc yes, all the qemu serial console is identified with source=qemu by `cc-proxy`

File: docs/debug-agent.md (hunk 1 above)
Comment (jcvenegas) [inline docs/debug-agent.md]: Changed :)

File: docs/debug-agent.md (hunk 2 above)
Comment (jcvenegas) [inline docs/debug-agent.md]: ups wrong place,  I wanted to have `cc-proxy`,  thanks for notice it , fixed now.

File: docs/debug-agent.md (hunk 3 above)
Comment (jcvenegas) [inline docs/debug-agent.md]: Agree, changed to /etc/systemd/system/cc-proxy.service.d/proxy-debug.conf

File: docs/debug-agent.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+[Clear Containers Agent](https://github.com/clearcontainers/agent):
+
+The Clear Containers Agent relies heavily on libcontainer used by runc
+(standard to run containers on bare metal).
+
+To provide a debug log of any agent activity on a guest, the Clear Containers
+agent sends logs through a qemu serial console that are collected by `cc-proxy`
+and shown in its logs. By default, this agent logs are not collected by 
+[cc-proxy](https://github.com/clearcontainers/proxy) but can be enabled with a
+simple change to service file.
```
Comment (devimc) [inline docs/debug-agent.md]: well, now you are not changed the service file, you are creating a configuration file for the service

Comment (jcvenegas) [inline docs/debug-agent.md]: true :P, updated.

**Comment by jcvenegas** [comment]
hi @iphutch, could you please take a look to this PR ?

File: docs/debug-agent.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+systemctl daemon-reload
+systemctl restart cc-proxy
+```
+2- Run a container to generate agent debug logs
+
+```
+sudo docker run -ti busybox true
+```
+3- Filter agent debug logs from cc-proxy logs
+`cc-proxy` show the source where is collating information, lets
```
Comment (iphutch) [inline docs/debug-agent.md]: I was confused by this sentence, is this correct:
The `cc-proxy` logs show the sources of its collated information. To see the agent debug logs, filter `cc-proxy` by the QEMU serial console the agent logs were sent through. The QEMU serial console is represented by `source=qemu`.

File: docs/debug-agent.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Get Clear Containers Agent debug logs
+
+When a new pod is created Clear Containers runtime launch a special VM which
```
Comment (jcvenegas) [inline docs/debug-agent.md]: Changed.

File: docs/debug-agent.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+uses a VM-agent. The VM-agent allows to spawn processes on behalf of
+pod/container(s) running inside the virtual machine. 
+
+From Clear Containers 3.0-beta release a new agent is used,
+[Clear Containers Agent](https://github.com/clearcontainers/agent):
+
+The Clear Containers Agent relies heavily on libcontainer used by runc
+(standard to run containers on bare metal).
```
Comment (jcvenegas) [inline docs/debug-agent.md]: Changed by the second, just changed to standard *program* instead of practice

File: docs/debug-agent.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+To provide a debug log of any agent activity on a guest, the Clear Containers
+agent sends logs through a qemu serial console that are collected by [cc-proxy](https://github.com/clearcontainers/proxy)
```
Comment (jcvenegas) [inline docs/debug-agent.md]: Changed

File: docs/debug-agent.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+and shown in its logs. By default, this agent logs are not collected by
```
Comment (jcvenegas) [inline docs/debug-agent.md]: Changed.

File: docs/debug-agent.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+[Clear Containers Agent](https://github.com/clearcontainers/agent):
+
+The Clear Containers Agent relies heavily on libcontainer used by runc
+(standard to run containers on bare metal).
+
+To provide a debug log of any agent activity on a guest, the Clear Containers
+agent sends logs through a qemu serial console that are collected by [cc-proxy](https://github.com/clearcontainers/proxy)
+and shown in its logs. By default, this agent logs are not collected by 
+`cc-proxy` but can be enabled by adding
+`-log debug` option to `cc-proxy`.
```
Comment (jcvenegas) [inline docs/debug-agent.md]: Added

File: docs/debug-agent.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+1- Add `-log debug` option to  `cc-proxy`
```
Comment (jcvenegas) [inline docs/debug-agent.md]: Added

File: docs/debug-agent.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+cat << EOT >  /etc/systemd/system/cc-proxy.service.d/proxy-debug.conf
+[Service]
+ExecStart=
+ExecStart=/usr/libexec/clear-containers/cc-proxy -log debug
+EOT
+# Restart cc-proxy to provide debug logs.
+systemctl daemon-reload
+systemctl restart cc-proxy
+```
+2- Run a container to generate agent debug logs
```
Comment (jcvenegas) [inline docs/debug-agent.md]: Added

File: docs/debug-agent.md (hunk 6 above)
Comment (jcvenegas) [inline docs/debug-agent.md]: Hi, sorry for not being clear, almost, just one clarification: There is only one QEMU serial console and  is used to send the agent logs. So I changed a bit, please take a look.

**Comment by jcvenegas** [comment]
@iphutch thanks for the review, please review the new changes.

File: docs/debug-agent.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+```
+2- Run a container to generate the agent debug logs
+
+```
+sudo docker run -ti busybox true
+```
+3- Filter the agent debug logs from the `cc-proxy` logs
+The `cc-proxy` logs show the sources of its collated information. To only see
+the agent debug logs, filter `cc-proxy` logs by the QEMU serial console (the
+agent logs were sent through it ). The QEMU serial console is represented by
```
Comment (iphutch) [inline docs/debug-agent.md]: remove space before closing parenthesis.

Comment (jcvenegas) [inline docs/debug-agent.md]: @iphutch  thank you! removed

**Pull request merged**